src/main/java/org/datadog/jenkins/plugins/datadog/configuration/api/intake/DatadogIntakeUrls.java

```diff
@@ -39,6 +40,17 @@ public DatadogIntakeUrls(String apiUrl, String logsUrl, String webhooksUrl) {
         this.webhooksUrl = webhooksUrl;
     }
 
+    /**
+     * Invoked by XStream when this object is deserialized.
+     * Ensures environment variables have higher priority than configuration persisted on disk
+     */
+    protected Object readResolve() {
+        String apiUrl = System.getenv().getOrDefault(TARGET_API_URL_PROPERTY, this.apiUrl);
+        String logsUrl = System.getenv().getOrDefault(TARGET_LOG_INTAKE_URL_PROPERTY, this.logsUrl);
+        String webhooksUrl = System.getenv().getOrDefault(TARGET_WEBHOOK_INTAKE_URL_PROPERTY, this.webhooksUrl);
```

Comment:
```suggestion
       Map<String, String> envVars = System.getenv()
        String apiUrl = envVars.getOrDefault(TARGET_API_URL_PROPERTY, this.apiUrl);
        String logsUrl = envVars.getOrDefault(TARGET_LOG_INTAKE_URL_PROPERTY, this.logsUrl);
        String webhooksUrl = envVars.getOrDefault(TARGET_WEBHOOK_INTAKE_URL_PROPERTY, this.webhooksUrl);
```

Should we generate a local variable to avoid calling `System.getenv()` repeteadly?

Reply:
Makes sense, fixed